kinder/ci/workflows/external-ca-tasks.yaml

```diff
@@ -0,0 +1,234 @@
+
+        # Remove the ca.key from all Nodes.
+        # The KCM tolerates the --cluster-signing-key flag, pointing to a file that is missing.
+        docker exec $CLUSTER-control-plane-1 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-control-plane-2 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-worker-1 rm /etc/kubernetes/pki/ca.key
+        docker exec $CLUSTER-worker-2 rm /etc/kubernetes/pki/ca.key
+  - name: init
+    description: |
+      Initializes the Kubernetes cluster with version "kubernetesVersion"
+      by starting the bootstrap control-plane node
+    cmd: kinder
+    args:
+      - do
+      - kubeadm-init
+      - --name={{ .vars.clusterName }}
+      - --loglevel=debug
+      - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+    timeout: 5m
+  - name: join
+    description: |
+      joins a secondary control-plane and worker Nodes to the cluster
+    timeout: 5m
+    cmd: /bin/bash
+    args:
+      - -c
+      - |
+        #!/bin/bash
+        # Enable output and exit on errors
+        set -o
+        set -e
+
+        # Setup variables. Having the explicit dockershim socket assumes the kinder base image has Docker.
+        CLUSTER={{ .vars.clusterName }}
+        FLAGS="--ignore-preflight-errors=all --cri-socket=/var/run/dockershim.sock --v={{ .vars.kubeadmVerbosity }}"
+        NODES=($CLUSTER-control-plane-2 $CLUSTER-worker-1 $CLUSTER-worker-2)
+
+        # Join the secondary CP node and the workers by using the kubelet.conf as discovery file
+        docker exec $CLUSTER-control-plane-2 kubeadm join --control-plane --discovery-file=/etc/kubernetes/kubelet.conf $FLAGS
+        docker exec $CLUSTER-worker-1 kubeadm join --discovery-file=/etc/kubernetes/kubelet.conf $FLAGS
+        docker exec $CLUSTER-worker-2 kubeadm join --discovery-file=/etc/kubernetes/kubelet.conf $FLAGS
+
+        # Wait for these remaining Nodes to be Ready
+        set +x
+        for node in "${NODES[@]}"
+        do
+          while [[ $(docker exec $CLUSTER-control-plane-1 kubectl --kubeconfig=/etc/kubernetes/admin.conf get node $node -o=json | jq -r '.status.conditions[] | select(.type | contains("Ready")) | .status') != "True" ]]
+          do
+            echo "Waiting for Node "$node" to become Ready..." && sleep 1
+          done
+        done
+  - name: e2e-kubeadm
+    description: |
+      Runs kubeadm e2e test
+    cmd: kinder
+    args:
+      - test
+      - e2e-kubeadm
+      - --test-flags=--report-dir={{ .env.ARTIFACTS }} --report-prefix=e2e-kubeadm --ginkgo.dryRun=true
+      - --name={{ .vars.clusterName }}
+      - --loglevel=debug
+  - name: e2e
+    description: |
+      Runs Kubernetes e2e test (conformance)
+    cmd: kinder
+    args:
+      - test
+      - e2e
+      - --test-flags=--report-dir={{ .env.ARTIFACTS }} --report-prefix=e2e --ginkgo.dryRun=true
+      - --parallel
+      - --name={{ .vars.clusterName }}
+      - --loglevel=debug
+    timeout: 35m
```

Comment:
Given that the focus of this test is to validate init and join workflow with external CA, IMO the conformance step could be skipped, wdyt?

Reply:
I think it is fine to not include the conformance run here.

Reply:
updated.